streaming/base/format/base/reader.py

```diff
@@ -74,6 +75,152 @@ def __len__(self) -> int:
         """
         return self.samples
 
+    def _evict_raw(self) -> None:
+        """Remove all raw files belonging to this shard."""
+        for raw_info, _ in self.file_pairs:
+            filename = os.path.join(self.dirname, self.split, raw_info.basename)
+            if os.path.exists(filename):
+                os.remove(filename)
```

Comment:
Will there be a case where more than one thread/process is trying to delete a file? could be a edge case of race condition here where they pass line 82 !!! Thinking to either wrap `os.remove` in a `try..except` block where we ignore exception in `except` part or use `shutil.rmtree(filename, ignore_errors=True)` !

Reply:
This is only called behind the lock `StreamingDataset._cache_filelock`

`StreamingDataset.evict_shard` -> takes lock -> `StreamingDataset._evict_shard` -> `Reader.evict` -> `Reader._evict_raw`

`StreamingDataset.download_shard` -> takes lock -> `StreamingDataset._evict_coldest_shard` -> `StreamingDataset._evict_shard` -> `Reader.evict` -> `Reader._evict_raw`